src/date_input.jsx

```diff
@@ -86,7 +87,20 @@ var DateInput = React.createClass({
   },
 
   render () {
-    const { date, locale, minDate, maxDate, excludeDates, includeDates, filterDate, dateFormat, onChangeDate, ...rest } = this.props // eslint-disable-line no-unused-vars
+    const { customActivator, date, locale, minDate, maxDate, excludeDates, includeDates, filterDate, dateFormat, onChangeDate, ...rest } = this.props // eslint-disable-line no-unused-vars
+
+    if (customActivator) {
+      const activatorElement = React.cloneElement(customActivator, {
+        ...rest,
+        ref: 'input',
```

Comment:
Because of this ref DateInput component will call `this.refs.input.focus` on your `customActivator` if its `focus` method is called, might be worth adding a check to see if the method's defined.